### ISSUE TITLE
Recent Issue with 0.4.2: Crash when requiring package: "double free or corruption (fasttop), Aborted"

### ISSUE THREAD

**Issue by ghost** [node-webrtc/node-webrtc#588]
Operating System: Debian GNU/Linux 10 (buster), Virtualbox guest
Kernel: 4.19.0-6-amd64
Node Version: 12.14.0 (verified with `node --version`)
`node-webrtc` Version: 0.4.2 (verified with `npm show wrtc version`)

I have been using `node-webrtc` within a project for several weeks without issue.
As of yesterday, an error message is output to stdout when I run the following script to require the package.

> const wrtc = require('wrtc');

That's the script, just a single line with the require statement.
It outputs this message to the console:

> double free or corruption (fasttop)
> Aborted

I originally encountered this on an Arch Linux install. Thinking it might be something on my end, I set up a minimalist Debian VM, installed Node 12.14.0 (the current LTS version), and tried to run the same one line script to require the package. I got the same output on both operating systems. Curious if other people are having the same problem.

Any help would be greatly appreciated :)

**Comment by ghost** [comment]
After updating from `node-webrtc` version `0.4.2` to `0.4.3` (using `yarn remove wrtc`, `yarn add wrtc@0.4.3`), this error goes away.

However, if I revert to version `0.4.2` from `0.4.3`, the error still occurs.
For my use case, I'm just going to use version `0.4.3`, but `0.4.2` still seems broken , possibly due to changes meant for version `0.4.3`.

**Comment by markandrus** [comment]
@3th3r3um good to know things are improved with `0.4.3`, although I'm still concerned that apparently an issue arose with `0.4.2` out of the blue:

> I have been using `node-webrtc` within a project for several weeks without issue.

**Comment by markandrus** [comment]
Actually, is it possible you updated Node in the meantime? I observed some changes in the Node 12 branch that started leading to crashes. This was also happening in 13. Those are fixed in `0.4.3`.

**Comment by ghost** [comment]
I'm using nvm (Node version manager) to manage Node versions, and I can confirm that I'm using and have been using Node 12.14.0 LTS.

Running back on Arch Linux again, I just created a new project now to confirm. Installing and requiring wrtc@0.4.2 results in the "double free or corruption (fasttop)" message and a core dump.  Uninstalling it and installing and requiring wrtc@0.4.3 works fine with no errors.

I also tried both 0.4.2 and 0.4.3 with Node 13.5.0 (the current release). 0.4.3 works without any errors, 0.4.2 results in a core dump (although it does not include the "double free or corruption (fasttop)" error message that I saw when using 12.14.0). Although that could just be because there is no official support for 0.4.2 with Node 13.

To reproduce for wrtc@0.4.2 on Node v12.14.0 (LTS):

```
mkdir issue-588
cd issue-588
nvm install 12.14.0
nvm use 12.14.0
node --version # (should be v12.14.0)
yarn init
yarn add wrtc@0.4.2
echo "require('wrtc');" > index.js
node index.js
```

Which should result in:

> double free or corruption (fasttop)
> [1]    270847 abort (core dumped)  node index.js

To reproduce for wrtc@0.4.3 on Node v12.14.0 (LTS):

```
mkdir issue-588
cd issue-588
nvm install 12.14.0
nvm use 12.14.0
node --version # (should be v12.14.0)
yarn init
yarn add wrtc@0.4.3
echo "require('wrtc');" > index.js
node index.js
```

Which should run and exit without any errors.
The only difference between the two being the version of wrtc.

**Comment by markandrus** [comment]
Thanks for your reproduction steps. The change I introduced between wrtc 0.4.2 and 0.4.3 is [this one](https://github.com/node-webrtc/node-webrtc/pull/580/commits/e1cdecb704d26891a028ed16f01d468c71427692). I didn't bother to dig much deeper into exactly _what_ changed in Node/N-API, but at some point in the Node 12.x release family, something N-API related changed which meant I need to delay invoking a constructor (AsyncContextReleaser in this case, a somewhat special class I introduced meant to work around some other funny N-API business).
